src/pipeline.cc

```diff
@@ -279,23 +279,6 @@ class PipelineWorker : public Nan::AsyncWorker {
         }
         xfactor = static_cast<double>(shrunkOnLoadWidth) / static_cast<double>(targetResizeWidth);
         yfactor = static_cast<double>(shrunkOnLoadHeight) / static_cast<double>(targetResizeHeight);
-        xshrink = std::max(1, static_cast<int>(floor(xfactor)));
```

Comment:
Can we remove the `xshrink`, `yshrink`, `xresidual` and `yresidual` variables entirely now?

Reply:
Thats what I wanted to do. But they are still used outside of the resize logic for the shrink_on_load logic.
Particularly xresidual and yresidual is used here 
https://github.com/lovell/sharp/blob/a3457a0b68cff73c9a502e75e28aeb7c074bf157/src/pipeline.cc#L245  this reduce the shrink_on_load.

and xshrink is also used in context of shrink_on_load here https://github.com/lovell/sharp/blob/a3457a0b68cff73c9a502e75e28aeb7c074bf157/src/pipeline.cc#L226